### PR TITLE
Actually check connection errors

### PR DESCRIPTION
**Pull request by otoolep**
Fix issue #4463

### CODE REVIEW

**Comment by otoolep** [comment]
With actual error checking, a helpful response is returned:

```
$ influx                                                                                                                                          
Failed to connect tohttp://localhost:8086
Please check your connection settings and ensure 'influxd' is running. 
```

**Comment by otoolep** [comment]
@beckettsean @corylanou

**Comment by beckettsean** [comment]
Nice catch, much improved!

File: cmd/influx/main.go [hunk 1]
```diff
@@ -158,7 +158,10 @@ Examples:
 	}
 
 	if err := c.connect(""); err != nil {
-
```
Comment (otoolep) [inline cmd/influx/main.go]: Actually do something if an error is returned, don't just carry on.

**Comment by jwilder** [comment]
:+1:

**Comment by gunnaraasen** [comment]
:+1:

File: cmd/influx/main.go (hunk 1 above)
Comment (corylanou) [inline cmd/influx/main.go]: what???  lol, crazy!  I think code reviews failed us the last time.  Nice catch.

**Comment by corylanou** [comment]
+1

**Pull request merged**